src/modm/ui/menu/standard_menu_impl.hpp

```diff
@@ -11,67 +12,68 @@
  */
 // ----------------------------------------------------------------------------
 
-#include "standard_menu.hpp"
+#ifndef MODM_STANDARD_MENU_HPP
+#	error 	"Don't include this file directly, use standard_menu.hpp instead!"
+#endif
 
-modm::MenuEntry::MenuEntry(const char* text, uint16_t space, MenuEntryCallback func):
+template<typename Allocator>
+modm::MenuEntry<Allocator>::MenuEntry(const char* text, uint16_t space, MenuEntryCallback<Allocator> func):
 	text(text, space),
 	callback(func)
 {
 }
 
-modm::StandardMenu::StandardMenu(modm::ViewStack* stack, uint8_t identifier) :
-	modm::AbstractMenu(stack, identifier),
+template<typename Allocator>
+modm::StandardMenu<Allocator>::StandardMenu(modm::ViewStack<Allocator>* stack, uint8_t identifier) :
+	modm::AbstractMenu<Allocator>(stack, identifier),
 	display_update_time(500),
 	timer(std::chrono::milliseconds(display_update_time)),
 	buttonAction(false),
 	title(""),
 	homePosition(0),
 	position(0)
 {
-	this->maximalDrawnEntrys = (getViewStack()->getDisplay().getHeight()- 16) / 8 ;
+	this->maximalDrawnEntrys = (this->getViewStack()->getDisplay().getHeight()- 16) / 8 ;
 }
 
-modm::StandardMenu::StandardMenu(modm::ViewStack* stack, uint8_t identifier, const char* title) :
-	modm::AbstractMenu(stack, identifier),
+template<typename Allocator>
+modm::StandardMenu<Allocator>::StandardMenu(modm::ViewStack<Allocator>* stack, uint8_t identifier, const char* title) :
+	modm::AbstractMenu<Allocator>(stack, identifier),
 	display_update_time(500),
 	timer(std::chrono::milliseconds(display_update_time)),
 	buttonAction(false),
 	title(title),
 	homePosition(0),
 	position(0)
 {
-	this->maximalDrawnEntrys = (getViewStack()->getDisplay().getHeight()- 16) / 8 ;
+	this->maximalDrawnEntrys = (this->getViewStack()->getDisplay().getHeight()- 16) / 8 ;
 }
 
-modm::StandardMenu::~StandardMenu()
+template<typename Allocator> void
+modm::StandardMenu<Allocator>::addEntry(const char* text, MenuEntryCallback<Allocator> func)
 {
-}
-
-void
-modm::StandardMenu::addEntry(const char* text, MenuEntryCallback func)
-{
-	modm::MenuEntry entry(text, (getViewStack()->getDisplay().getWidth()-16)/6, func);
+	modm::MenuEntry entry(text, (this->getViewStack()->getDisplay().getWidth()-16)/6, func);
 	this->entries.append(entry);
 }
 
-void
-modm::StandardMenu::setTitle(const char* text)
+template<typename Allocator> void
+modm::StandardMenu<Allocator>::setTitle(const char* text)
 {
 	this->title = text;
 }
 
 
-void
-modm::StandardMenu::draw()
+template<typename Allocator> void
+modm::StandardMenu<Allocator>::draw()
 {
-	modm::GraphicDisplay* display = &getViewStack()->getDisplay();
+	typename modm::GraphicDisplay* display = &this->getViewStack()->getDisplay();
```

Comment:
I'm confused what `typename` is used here for.

Reply:
The gist is that since `getViewStack` returns a template object, we need to add typename before the type so the compiler knows this is actually a `modm::GraphicDisplay`.

Read a complete answer here: https://stackoverflow.com/questions/610245/where-and-why-do-i-have-to-put-the-template-and-typename-keywords